src/pilot/assign_role.py

```diff
@@ -1160,10 +1165,26 @@ def select_os_volume(os_volume_size_gb, ironic_client, drac_client, node_uuid):
 
                     break
 
+            # Note: This code block represents single disk scenario.
             if raid_size_gb == 0:
-                raise RuntimeError("There must be either a virtual disk that "
-                                   "is not a RAID 0 to install the OS on, or "
-                                   "os-volume-size-gb must be specified")
+                try:
```

Comment:
I think we should remove this try/except block.  If we just let any exception that occurs when communicating with the iDRAC go up the stack, then it will be caught in main() and handled correctly.